code/modules/vending/vending.dm

```diff
@@ -666,15 +666,15 @@ ADMIN_INTERACT_PROCS(/obj/machinery/vending, proc/throw_item, proc/admin_command
 							qdel(product)
 						product.product_amount--
 					if(src.pay && vended)
+						var/obj/machinery/vending/player/vMachine = src
 						if (src.acceptcard && account)
 							account["current_money"] -= product.product_cost
 						else
 							src.credit -= product.product_cost
-						if (!player_list)
+						if (!player_list || !vMachine.owneraccount)
```

Comment:
Is there a guarantee that `src` is a player vending machine at this point? Ie can this code be reached while operating a normal vending machine? If it can then this will runtime.

Reply:
player_list should be null on anything that's not a player vending machine - normal vending machines use product_list - so my understanding was the check would never reach !vMachine.owneraccount if it wasn't a player vending machine.

That being said ?. are free so